SEMain/src/lib/PluginConfig.cpp

```diff
@@ -47,6 +47,7 @@ std::map<std::string, Configuration::OptionDescriptionList> PluginConfig::getPro
 void PluginConfig::initialize(const UserValues& args) {
   m_plugin_path = args.at(PLUGIN_DIRECTORY).as<std::string>();
   m_plugin_list = args.at(PLUGIN).as<std::vector<std::string>>();
+  m_plugin_list.erase(std::remove(m_plugin_list.begin(), m_plugin_list.end(), ""), m_plugin_list.end());
```

Comment:
I don't understand this line.

I guess the intent is to remove all empty strings from the list, but as I understand it std::remove already does that and returns and end() iterator making the erase redundant?

Am I misunderstanding this function?

Reply:
Nevermind I understand, remove doesn't resize the array and erase cleans that up. So it's correct but it's rather cryptic.